src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -217,6 +263,216 @@ await contextualFilter.EvaluateAsync(context, appContext).ConfigureAwait(false)
             return enabled;
         }
 
+        public ValueTask<Variant> GetVariantAsync(string feature, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrEmpty(feature))
+            {
+                throw new ArgumentNullException(nameof(feature));
+            }
+
+            return GetVariantAsync(feature, null, false, cancellationToken);
+        }
+
+        public ValueTask<Variant> GetVariantAsync(string feature, TargetingContext context, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrEmpty(feature))
+            {
+                throw new ArgumentNullException(nameof(feature));
+            }
+
+            return GetVariantAsync(feature, context, true, cancellationToken);
+        }
+
+        private async ValueTask<Variant> GetVariantAsync(string feature, TargetingContext context, bool useContext, CancellationToken cancellationToken)
+        {
+            FeatureDefinition featureDefinition = await _featureDefinitionProvider
+                .GetFeatureDefinitionAsync(feature)
+                .ConfigureAwait(false);
+
+            if (featureDefinition == null)
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.MissingFeature,
+                    $"The feature declaration for the feature '{feature}' was not found.");
+            }
+
+            if (featureDefinition.Allocation == null)
```

Comment:
I think the overarching sentiment for variants is to return null rather than throw for situations like this.

Reply:
Ok, so only for when things are missing from the definition right? Still throw in situations like where both `ConfigurationValue` and `ConfigurationReference` are specified?